fc/parsing/actions.py

```diff
@@ -965,19 +964,18 @@ def _add_or_replace_equations(self):
             lhs = eq.lhs
             if lhs.is_Derivative:
                 var = lhs.args[0]
-                term = self.model.get_ontology_terms_by_variable(var)[0]
 
                 # Initial value must be set via inputs, or already be set (if this was already a state variable)
-                if var.initial_value is None and term not in initial_values:
+                if var.initial_value is None and var not in initial_values:
+                    term = self.model.get_ontology_terms_by_variable(var)[0]
```

Comment:
When looping over self.initial_values to create the var -> init map, you could also create a var -> term map to ensure the term the user specified is given in the error message.

Reply:
The reason being, a variable could be annotated by multiple different terms, possibly from different ontologies.

Reply:
But the term is only used for variables that are _not_ in `self.initial_values`!

Reply:
Good point!